src/inference/src/threading/ie_istreams_executor.cpp

```diff
@@ -235,12 +321,91 @@ Parameter IStreamsExecutor::Config::GetConfig(const std::string& key) const {
+
+    config._small_core_offset = num_big_cores_phys * 2;
+    // if nthreads is set, calculate hybrid aware parameters by nthreads
+    if (config._threads) {
+        // limit threads to reasonable range
+        auto threads = config._threads > num_cores_phys ? num_cores_phys : config._threads;
+        // all threads are placed on big cores
+        if (threads <= num_big_cores_phys) {
+            config._big_core_streams = config._streams;
+            config._threads_per_stream_big = std::max(1, threads / config._streams);
+        } else {
+            // big cores first, then small cores
+            // big_streams = all_streams * big_cores / base_streams
+            config._big_core_streams =
+                config._streams * num_big_cores_phys % base_streams_total == 0
+                    ? config._streams * num_big_cores_phys / base_streams_total
+                    : (config._streams * num_big_cores_phys + base_streams_total) / base_streams_total;
+            config._small_core_streams = config._streams - config._big_core_streams;
+            config._threads_per_stream_big = std::max(1, num_big_cores_phys / config._big_core_streams);
+            if (config._small_core_streams > 0) {
+                // Normally, threads_per_stream_small = threads_per_stream_big * 2, but special conditions are as
+                // follows: Pcore=8, Ecore=4, custom set nthreads=10, nstreams=10, calculate result: big_streams=8,
+                // small_streams=2. There are only 2 threads left, threads_per_stream_small can only be 1.
+                const int remain_threads = threads - config._threads_per_stream_big * config._big_core_streams;
+                config._threads_per_stream_small =
+                    std::min(remain_threads / config._small_core_streams, config._threads_per_stream_big * 2);
+            }
+        }
+    } else {  // user not set nthreads
+        config._threads_per_stream_big =
+            std::min(num_big_cores_phys, std::max(1, base_streams_total / config._streams));
+        config._big_core_streams = std::min(config._streams, num_big_cores_phys / config._threads_per_stream_big);
+        config._small_core_streams = config._streams - config._big_core_streams;
+        config._threads_per_stream_small = config._small_core_streams > 0 ? config._threads_per_stream_big * 2 : 0;
+        // The number of threads on small core exceeds the number of small core due to threads indivisible on big core.
+        // We should add the number of threads on big core, reduce the one on small core.
+        if (num_small_cores < config._threads_per_stream_small) {
+            config._big_core_streams += 1;
+            config._small_core_streams -= 1;
+            config._threads_per_stream_big = std::max(1, num_big_cores_phys / config._big_core_streams);
+            config._threads_per_stream_small = config._small_core_streams > 0 ? config._threads_per_stream_big * 2 : 0;
```

Comment:
_threads_per_stream_small is same as _threads_per_stream_big. Here should be config._small_core_streams > 0 ? config._threads_per_stream_big : 0